src/app/tests/TestReadInteraction.cpp

```diff
@@ -2628,10 +2581,10 @@ void TestReadInteraction::TestSubscribeWildcard(nlTestSuite * apSuite, void * ap
         // the initial list followed by additional items.
         constexpr size_t kExpectedAttributeResponse = 29 * 2 + (kMockAttribute4ListLength - 4) + (kMockAttribute4ListLength - 4);
 #endif
-        NL_TEST_ASSERT(apSuite, delegate.mNumAttributeResponse == kExpectedAttributeResponse);
-        NL_TEST_ASSERT(apSuite, delegate.mNumArrayItems == 12);
-        NL_TEST_ASSERT(apSuite, engine->GetNumActiveReadHandlers(ReadHandler::InteractionType::Subscribe) == 1);
-        NL_TEST_ASSERT(apSuite, engine->ActiveHandlerAt(0) != nullptr);
+        EXPECT_EQ((unsigned) delegate.mNumAttributeResponse, kExpectedAttributeResponse);
+        EXPECT_EQ(delegate.mNumArrayItems, 12);
+        EXPECT_EQ(engine->GetNumActiveReadHandlers(ReadHandler::InteractionType::Subscribe), 1u);
+        EXPECT_NE(engine->ActiveHandlerAt(0), nullptr);
```

Comment:
I'll stop individually marking these, but i think most of the cases where we have `EXPECT_NE\(.*nullptr\);` are going to be testing pointers that we're about to dereference

Reply:
done for many  occurances